### PR TITLE
Add upstage document reader

### PR DESCRIPTION
**Pull request by junkeon**
# Description

This PR introduces a new document reader tailored for the Upstage API, leveraging the Layout-Analyzer Model. It encompasses essential functionalities for document retrieval, response parsing, and management of various split types.

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [x] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [x] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: llama-index-integrations/readers/llama-index-readers-file/llama_index/readers/file/upstage/base.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+            files = {"document": open(file_path, "rb")}
+            response = requests.post(self.api_base, headers=headers, files=files)
+        except requests.exceptions.RequestException as e:
+            raise ValueError(f"API call error: {e}")
+        finally:
+            files["document"].close()
+
+        return response.json()
+
+    def load(
```
Comment (logan-markewich) [inline llama-index-integrations/readers/llama-index-readers-file/llama_index/readers/file/upstage/base.py:114]: I believe this method should be `load_data(file_path, ..., **kwargs)` ? Double check the base class 👁️ https://github.com/run-llama/llama_index/blob/161405793c8827f9d3ff5acc98906d06b408f27d/llama-index-core/llama_index/core/readers/base.py#L35

**Review by logan-markewich — APPROVED** [review]
LGTM, but lets implement the `load_data()` method instead

**Comment by junkeon** [comment]
@logan-markewich Thank you for your feedback and approval. I have pushed the changes with the renamed function as per your suggestion. I will ensure that the code passes the tests before merging it.

File: llama-index-integrations/readers/llama-index-readers-file/llama_index/readers/file/upstage/base.py [hunk 2]
```diff
@@ -0,0 +1,314 @@
+import io
+import os
+from pathlib import Path
+from typing import Dict, List, Union
+
+import fitz
+import requests
+from llama_index.core.readers.base import BaseReader
+from llama_index.core.schema import Document
+from llama_index.legacy.llms.generic_utils import get_from_param_or_env
```
Comment (logan-markewich) [inline llama-index-integrations/readers/llama-index-readers-file/llama_index/readers/file/upstage/base.py:10]: Ah I missed this. Should be `from llama_index.core.base.llms.generic_utils import get_from_param_or_env`

**Comment by junkeon** [comment]
After a thorough internal review, I'll resend the PR for your consideration, and I truly appreciate your patience and input throughout this process.

**Pull request closed without merging**